pkg/cli/image/manifest/manifest.go

```diff
@@ -434,28 +435,29 @@ func ProcessManifestList(ctx context.Context, srcDigest digest.Digest, srcManife
 			klog.V(5).Infof("Filtered manifest list to new digest %s:\n%s", manifestDigest, body)
 		}
 
-		for i, manifest := range t.Manifests {
+		for i, manifest := range filtered {
 			childManifest, err := manifests.Get(ctx, manifest.Digest, distribution.WithManifestMediaTypes([]string{manifestlist.MediaTypeManifestList, schema2.MediaTypeManifest}))
 			if err != nil {
 				return nil, nil, "", fmt.Errorf("unable to retrieve source image %s manifest #%d from manifest list: %v", ref, i+1, err)
 			}
-			srcManifests = append(srcManifests, childManifest)
+			childManifests = append(childManifests, childManifest)
 		}
 
 		switch {
-		case len(srcManifests) == 1 && !keepManifestList:
-			manifestDigest, err := registryclient.ContentDigestForManifest(srcManifests[0], srcDigest.Algorithm())
+		case len(childManifests) == 1 && !keepManifestList:
+			// Just return the single platform specific image
+			manifestDigest, err := registryclient.ContentDigestForManifest(childManifests[0], srcDigest.Algorithm())
 			if err != nil {
 				return nil, nil, "", err
 			}
 			klog.V(2).Infof("Chose %s/%s manifest from the manifest list.", t.Manifests[0].Platform.OS, t.Manifests[0].Platform.Architecture)
-			return srcManifests, srcManifests[0], manifestDigest, nil
+			return nil, childManifests[0], manifestDigest, nil
 		default:
-			return append(srcManifests, manifestList), manifestList, manifestDigest, nil
+			return childManifests, manifestList, manifestDigest, nil
 		}
 
 	default:
-		return []distribution.Manifest{srcManifest}, srcManifest, srcDigest, nil
+		return nil, srcManifest, srcDigest, nil
```

Comment:
Aren't we changing the behavior for cases where `keepManifestList` is false?

Reply:
Yes, but this PR also modifies the callers to handle that change.